daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Error.scala

```diff
@@ -137,28 +141,24 @@ object Error {
 
     // TODO https://github.com/digital-asset/daml/issues/9974
     //  get rid of Generic
-    final case class Generic(override val msg: String, detailMsg: String) extends Error
-    object Generic {
-      def apply(msg: String): Generic = Generic(msg, msg)
+    final case class Generic(override val msg: String) extends Error
+
+    final case class DamlException(error: interpretation.Error) extends Error {
+      // TODO https://github.com/digital-asset/daml/issues/9974
+      //  For now we try to preserve the exact same message (for the ledger API)
+      //  Review once all the errors are properly structured
+      override def msg: String = error match {
+        case interpretation.Error.ContractNotFound(cid) =>
+          // TODO https://github.com/digital-asset/daml/issues/9974
+          //   we should probably use ${cid.coid} instead of $cid
```

Comment:
Added this to https://github.com/digital-asset/daml/issues/9974 's description, feel free to delete [the comment](https://github.com/digital-asset/daml/issues/9974#issuecomment-867649444) if unnecessary. 🙂